ndspy/bmg.py

```diff
@@ -385,17 +392,22 @@ def save(self, encoding):
             data.extend(self.data)
             return data
 
-        def __repr__(self):
+        def __repr__(self) -> str:
             return f'{type(self).__name__}({self.type!r}, {self.data!r})'
 
-        def __str__(self):
+        def __str__(self) -> str:
             return f'[{self.type}:{self.data.hex()}]'
 
-    info = 0
-    stringParts = None
-    isNull = False
+    info: int = 0
+    stringParts: list[str] | None = None
```

Comment:
This can't be `None` -- `__init__()` accepts `None` but replaces it with `[]`, and then `save()` assumes that it's a list (see line 426)